code/obj/machinery/shield_generators/portable_shield_generator.dm

```diff
@@ -665,6 +673,7 @@ ADMIN_INTERACT_PROCS(/obj/machinery/shieldgenerator, proc/turn_on, proc/turn_off
 			var/obj/machinery/door/door = (locate() in src.loc)
 			if(door)
 				door.linked_forcefield = src
+				door.SubscribeToProcess() //necessary for power consumption while open
```

Comment:
This is quite a heavy performance hit if every single door on Nadir is a processing machine, consider making it only subscribe to process when it actually needs to (or just moving power usage to only when the field is created)

Reply:
Moving power usage to field creation is a little wonky, but if I can't think of a better way to dodge the processing, I'll probably end up doing that.

Reply:
Now subscribes to process only when door is open. Also converted poddoors to use base open and close along the way.